### DISCUSSION TITLE
I couldn't run a blank project on Android Studio on a Mac

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#1902] [Q&A] [answered]
So I followed the dev setup page and created a project through command line and tried to open the android project on Android Studio 2023.1.1, I get c/c++ Configuration problem.. and I couldn't run and test the even a blank project. 
I'm using the Axmol engine latest dev branch from github. Please what's wrong here..
![Screen Shot 2024-05-17 at 3 03 32 PM](https://github.com/axmolengine/axmol/assets/6400139/2a61d32a-c37d-450e-a89a-007f0b7b503f)

**Comment by rh101** [comment]
What is the exact command you used on the command line to generate the project (as in `axmol new` etc. etc.)?

**Reply by meehjeevan** [reply]
Like this 
axmol new project_name -p package_id -l cpp

**Comment by halx99** [comment]
> So I followed the dev setup page and created a project through command line and tried to open the android project on Android Studio 2023.1.1, I get c/c++ Configuration problem.. and I couldn't run and test the even a blank project. I'm using the Axmol engine latest dev branch from github. Please what's wrong here.. ![Screen Shot 2024-05-17 at 3 03 32 PM](https://private-user-images.githubusercontent.com/6400139/331541290-2a61d32a-c37d-450e-a89a-007f0b7b503f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5NTkyNjYsIm5iZiI6MTcxNTk1ODk2NiwicGF0aCI6Ii82NDAwMTM5LzMzMTU0MTI5MC0yYTYxZDMyYS1jMzdkLTQ1MGUtYTg5YS0wMDdmMGI3YjUwM2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTdUMTUxNjA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmJjMzgxMWUwYWQ5ZmM4MTI0NDc2ZTI4YzJjMzVhMmQzYWU0YzhkOTk5NjA4YmEzZWFjYmIzOTY0NTJhZDgwYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zSoEYXcIuUua_6NFzwoqAYWSCCZSs93qW8fUNYhCYWk)

seems the config probelm not affetct build, do you try build

**Reply by meehjeevan** [reply]
Build wasn't successful..  the errors was very long so I didn't post here.

**Reply by halx99** [reply]
could upload a text file

**Reply by halx99** [reply]
![image](https://github.com/axmolengine/axmol/assets/6977319/ef369f14-0046-4a30-80af-af7b75ca10a0)

I also have the config error on windows, but I can build APK

**Reply by halx99** [reply]
![image](https://github.com/axmolengine/axmol/assets/6977319/7843c0e4-d311-46a1-bbea-dc69764a6365)

**Reply by rh101** [reply]
> I also have the config error on windows, but I can build APK

Same for me, the config error always appears, but the build in unaffected, and completes successfully.

**Reply by paulocoutinhox** [reply]
Same, config has problems, but build works.

**Reply by meehjeevan** [reply]
ok plz review the way I installed cmake 3.28.1 manually.. 
I downloaded the cmake.app and extracted all files(not just bin folder) in android_sdk>cmake>3.28.1 folder like this plz check SC.
I also downloaded ninja separately and placed in bin folder.
Is it the correct way?
![Screen Shot 2024-05-18 at 11 11 11 AM](https://github.com/axmolengine/axmol/assets/6400139/8791db81-2330-4e86-8a15-b0fce4315d1c)
The errors points to the ninja related things. plz check the attached txt file for full errors.
[buildError.txt](https://github.com/axmolengine/axmol/files/15360077/buildError.txt)

**Reply by paulocoutinhox** [reply]
I generally install using brew:
https://brew.sh/

Then:
brew install cmake

**Reply by halx99** [reply]
your mac is arm64? the prebuilt glslcc only contains x64

**Reply by paulocoutinhox** [reply]
Mine is M1 (arm64)

**Reply by meehjeevan** [reply]
mine is x86_64 old mac(2013 model)

**Reply by halx99** [reply]
maybe your device is too old

**Reply by meehjeevan** [reply]
I tried installing latest cmake with home brew too.. but the issue is still same.. 
so my device is too old so that it's not working? the build was working in previous Axmol 2.1.0 version.
is the latest version of Axmol all working well on m2 or m3 macs?

**Reply by halx99** [reply]
the issue not caused by cmake, it's cause by glslcc can't run on your device, you can simply run tools/external/glslcc/glslcc to confirm

**Reply by halx99** [reply]
https://github.com/axmolengine/glslcc/releases

**Reply by halx99** [reply]
strange, axmol 2.1.0 use same glslcc release 1.9.4: https://github.com/axmolengine/axmol/blob/v2.1.0/1k/manifest.ps1#L11

**Reply by halx99** [reply]
```txt
dyld: Symbol not found: __ZTTNSt3__114basic_ofstreamIcNS_11char_traitsIcEEEE
  Referenced from: /Users/imac/Documents/tools/axmol/tools/external/glslcc/glslcc (which was built for Mac OS X 12.6)
  Expected in: /usr/lib/libc++.1.dylib
 in /Users/imac/Documents/tools/axmol/tools/external/glslcc/glslcc
/bin/sh: line 1:  1785 Abort trap: 6  
```

**Reply by meehjeevan** [reply]
My device is running on Mac os 10.15 7
But in error log glslcc is built for Mac os 12.6
So I must run it on mac os 12.6 ??

**Comment by halx99** [comment]
do you run glslcc directly? glslcc is opensource, and cmake based project, you can clone and build it

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by meehjeevan** [reply]
I don't know how to use it directly .. I was just trying test the blank project on android studio.

**Reply by halx99** [reply]
just run: `./tools/external/glslcc/glslcc` to check does it report dylib symbol error in axmol root

**Reply by halx99** [reply, marked as answer]
I have try to publish new release of `glslcc` to support macos-10.15, will info you once it done:
https://github.com/axmolengine/glslcc/commit/0fd2c087cdd01caba838fe07b8592d467f787bdb

**Reply by halx99** [reply]
Hi @meehjeevan , please update axmol to latest and try again

**Reply by meehjeevan** [reply]
again getting same errors
```
FAILED: /Users/imac/Documents/work/Axmol/LudoX/proj.android/app/build/runtime/axslc/cameraClear_fs 
  cd /Users/imac/Documents/work/Axmol/LudoX/proj.android/app/.cxx/Debug/4a1t483f/arm64-v8a/engine/axmol/core && /Users/imac/Documents/tools/axmol/tools/external/glslcc/glslcc --silent --err-format=msvc --lang=gles --profile=100 --automap --no-suffix "--defines=GLES2" --include-dirs=/Users/imac/Documents/tools/axmol/core/renderer/shaders --frag=/Users/imac/Documents/tools/axmol/core/renderer/shaders/cameraClear.frag --output=/Users/imac/Documents/work/Axmol/LudoX/proj.android/app/build/runtime/axslc/cameraClear_fs
  dyld: Symbol not found: __ZTTNSt3__114basic_ofstreamIcNS_11char_traitsIcEEEE
    Referenced from: /Users/imac/Documents/tools/axmol/tools/external/glslcc/glslcc (which was built for Mac OS X 12.6)
    Expected in: /usr/lib/libc++.1.dylib
   in /Users/imac/Documents/tools/axmol/tools/external/glslcc/glslcc
  /bin/sh: line 1:  6285 Abort trap: 6           /Users/imac/Documents/tools/axmol/tools/external/glslcc/glslcc --silent --err-format=msvc --lang=gles --profile=100 --automap --no-suffix "--defines=GLES2" --include-dirs=/Users/imac/Documents/tools/axmol/core/renderer/shaders --frag=/Users/imac/Documents/tools/axmol/core/renderer/shaders/cameraClear.frag --output=/Users/imac/Documents/work/Axmol/LudoX/proj.android/app/build/runtime/axslc/cameraClear_fs
  [627/1440] Compiling shader /Users/imac/Documents/tools/axmol/core/renderer/shaders/colorNormal.frag for ESSL100 ...
```

**Reply by halx99** [reply]
please run ./setup.ps1 again, then try again

**Reply by meehjeevan** [reply]
Finally it's working. Thanks a lot. 👍
